### ISSUE TITLE
Calling pre-commit --version as a module confuses it about its name

### ISSUE THREAD

**Issue by ssbarnea** [pre-commit/pre-commit#1273]
I found a minor bug in `pre-commit` which reports `__main__.py` instead of `pre-commit` when is asked about its version using module calling.

    python -m pre_commit --version
    __main__.py 1.20.0
                                                                                                                                                                                     
I would be nice to find a solution for it as I observed this issue on other projects too. Still, it seems that tools like pip, virtualenv, tox or pytest are not affected by this bug.

Calling tools as python modules is quite common on CI as it avoids two common issues: exec install path not in in PATH, wrong python interpreter would be used.

**Comment by asottile** [comment]
argparse by default infers the `prog` attribute from `sys.argv[0]` -- you can however pass hardcoded string to `ArgumentParser(prog=...)` to configure this

would you like to submit a patch?

**Comment by orcutt989** [comment]
Bombs on Python 3.7
```shell
macbook71:pre-commit user$ python -m pre_commit --version
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/user/Projects/pre-commit/pre_commit/__main__.py", line 3, in <module>
    from pre_commit.main import main
  File "/Users/user/Projects/pre-commit/pre_commit/main.py", line 8, in <module>
    import pre_commit.constants as C
  File "/Users/user/Projects/pre-commit/pre_commit/constants.py", line 26, in <module>
    VERSION = importlib_metadata.version('pre_commit')
  File "/usr/local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 509, in version
    return distribution(distribution_name).version
  File "/usr/local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 482, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/local/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 187, in from_name
    raise PackageNotFoundError(name)
importlib_metadata.PackageNotFoundError: pre_commit
```

**Comment by asottile** [comment]
works fine for me, how did you install pre-commit?

**Comment by asottile** [comment]
(it looks like you didn't, if you're developing on `pre-commit` make sure to `pip install -e .` or use `tox --devenv venv` to make a development environment)

**Comment by orcutt989** [comment]
🤦‍♂ @asottile My bad! Haha.

**Comment by orcutt989** [comment]
Submitting a patch in a moment...

**Comment by asottile** [comment]
via #1276

**Comment by asottile** [comment]
this has been released as part of [v2.0.0](https://github.com/pre-commit/pre-commit/releases/tag/v2.0.0) -- thanks again for the issue and patch!
